client/src/views/Feed/AddPost.vue

```diff
@@ -82,32 +95,75 @@ export default {
         postVideo: '',
         likes: 0,
         hasBeenLiked: false,
+        tag: [],
         caption: '',
         comments: []
       }
     }
   },
+  updated () {
+    if (!this.dialog && this.post.postVideo === "") {
+      this.$router.push('/feed')
+    }
+  },
   methods: {
     ...mapActions('posts', ['addPost']),
-    addToFeed () {
+    async addToFeed () {
+      let valueArr = this.$refs.textarea.value.split(" ")
+      valueArr.filter((word) => {
+        if (word.match(/#\w+/)) {
+          word = word.slice(1)
+          this.post.tag.push(word)
+        }
+      })
       this.post.caption = this.$refs.textarea.value
       this.addPost(this.post)
+      axios.post('/api/posts/save-post', this.post)
+        .then((response) => {
+        })
+        .catch(error => {
+          console.log(error)
+        })
       this.$router.push('/feed')
     },
-    uploadImage (e) {
+    async uploadVideo (e) {
       const files = e.target.files
       if (!files.length) return
-      const reader = new FileReader()
-      reader.readAsDataURL(files[0])
-      reader.onload = e => {
-        this.post.postVideo = e.target.result
-        this.$refs.video.src = this.post.postVideo
-        this.dialog = false
+      if (files[0].type.match(/video..../gi)) {
+        const reader = new FileReader()
+        reader.readAsDataURL(files[0])
+        const formData = new FormData()
+        formData.append('file', files[0])
+        try {
+          const videoUrl = await axios.post('https://emocean.dev/api/posts/upload-videos?type=single-video', formData, {
```

Comment:
You need to declare this url on higher level, for instance in api Service, because you might need in future multiple development environments or just url of api could change.
const API_URL = https://emocean.dev/api;
const uploadVideoUrl = API_URL  + '/upload-videos?type=single-video'
Move them outside of component - https://medium.com/canariasjs/vue-api-calls-in-a-smart-way-8d521812c322